pkg/receive/handler.go

```diff
@@ -470,6 +485,15 @@ func (h *Handler) fanoutForward(pctx context.Context, tenant string, replicas ma
 				ec <- errors.Wrapf(err, "get peer connection for endpoint %v", endpoint)
 				return
 			}
+
+			b, ok := h.backoffs[endpoint]
```

Comment:
I think this need to be mutexed, no? :thinking:

Reply:
Yes, you're super right. I've missed this one.